crates/chainspec/src/spec.rs

```diff
@@ -695,7 +695,7 @@ impl From<Genesis> for ChainSpec {
             chain: genesis.config.chain_id.into(),
             genesis,
             genesis_hash: None,
-            hardforks: ChainHardforks::new(hardforks),
+            hardforks: ChainHardforks::new(ordered_hardforks),
```

Comment:
im not too sure what this is even used for, cc @joshieDo

Reply:
I don't see any difference between using it or not but maybe there are exceptions that I don't have in mind. 

If this is not used, we can simply delete the whole block of code before:

```rust
// Uses ethereum or optimism main chains to find proper order
        #[cfg(not(feature = "optimism"))]
        let mainnet_hardforks: ChainHardforks = EthereumHardfork::mainnet().into();
        #[cfg(not(feature = "optimism"))]
        let mainnet_order = mainnet_hardforks.forks_iter();
        #[cfg(feature = "optimism")]
        let mainnet_hardforks = OptimismHardfork::op_mainnet();
        #[cfg(feature = "optimism")]
        let mainnet_order = mainnet_hardforks.forks_iter();

        let mut ordered_hardforks = Vec::with_capacity(hardforks.len());
        for (hardfork, _) in mainnet_order {
            if let Some(pos) = hardforks.iter().position(|(e, _)| **e == *hardfork) {
                ordered_hardforks.push(hardforks[pos].clone());
            }
        }
```

Reply:
We love lints :)